File: sample-services/npm/src/commands/add-something.ts
import { ICommand } from "../abstractions/command";
import { CommandHandlerBase } from "../abstractions/command-handler";

export interface IAddSomethingCommand extends ICommand {
    someNewData: string;
}

export class AddSomethingCommand implements IAddSomethingCommand {
    static readonly TYPE = "SampleServices.Commands.AddSomethingCommand";
    readonly type: string;
    someNewData: string;
    constructor(state?: Partial<IAddSomethingCommand>) {
        this.someNewData = "default";
        state && Object.assign(this, JSON.parse(JSON.stringify(state)));
        this.type = AddSomethingCommand.TYPE;
    }
}

export interface IAddSomethingHandler extends CommandHandlerBase<IAddSomethingCommand, number> { }

File: sample-services/dotnet/src/commands/add-something.cs
using Sample.Services.Abstractions;

namespace Sample.Services.Commands
{
    public class AddSomethingCommand : CommandBase, ICommand
    {
        public const string TYPE = nameof(AddSomethingCommand);
        public override string Type => TYPE;
        public string SomeNewData { get; set; }
        public AddSomethingCommand() 
        {
            SomeNewData = "default";
        }
    }

    public interface IAddSomethingCommandHandler : ICommandHandler<AddSomethingCommand, int> { }
}